core/src/main/java/org/elasticsearch/search/suggest/Suggest.java

```diff
@@ -387,7 +391,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         @SuppressWarnings("unchecked")
         public static Suggestion<? extends Entry<? extends Option>> fromXContent(XContentParser parser) throws IOException {
             ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.currentToken(), parser::getTokenLocation);
-            return XContentParserUtils.parseTypedKeysObject(parser, Aggregation.TYPED_KEYS_DELIMITER, Suggestion.class);
+            SetOnce<Suggestion> suggestion = new SetOnce<>();
```

Comment:
Tried that, Optional doesn't work well as a consumer though since it doesn't have a setter. That would somehow lead us back to the "original" signature in this PR where parseTypedKeysObject returns an Optional. Wouldn't mind going back there again.